src/engine.c

```diff
@@ -881,6 +881,24 @@ ENGINE_getKeyState(ENGINE* engine, char* keyName) {
   return state[scancode];
 }
 
+internal uint32_t
+ENGINE_findMouseCursorIndex(ENGINE* engine, char* cursorName) {
+  int cursorIndex;
+  for (int index = 0; index < 12; index++) {
+    char * name = ENGINE_MOUSE_CURSORS[index];
+    if (strcmp(name, cursorName) == 0) cursorIndex = index;
```

Comment:
This should probably be:
```c
internal int32_t
ENGINE_findMouseCursorIndex(ENGINE* engine, char* cursorName) {
  for (int index = 0; index < SDL_NUM_SYSTEM_CURSORS; index++) {
    char * name = ENGINE_MOUSE_CURSORS[index];
    if (STRINGS_EQUAL(cursorName, name)) {
      return index;
    }
  }
  return -1;
}
```

There's always the possibility that an invalid name is provided, so we need to be able to detect and handle that error case. (Note that I changed the returntype to be an `int32_t` so that we have an error option.)

Reply:
Currently if an unknown name is provided the loop does not change `cursorIndex` so it will return 0. So if you try to set the cursor to an unknown name it will set it back to the default arrow cursor. It's your call if you want the method to throw an error if that makes more sense tho.

Reply:
updated your changes in 8d7232c, still no error checking but SDL seems to take care of it either way.